go/vt/vtgate/planbuilder/testdata/reference_cases.json

```diff
@@ -121,8 +121,8 @@
           "Name": "main",
           "Sharded": false
         },
-        "FieldQuery": "select r1.col from ambiguous_ref_with_source as r1, ambiguous_ref_with_source where 1 != 1",
-        "Query": "select r1.col from ambiguous_ref_with_source as r1, ambiguous_ref_with_source",
+        "FieldQuery": "select r1.col from ambiguous_ref_with_source as r1 join ambiguous_ref_with_source where 1 != 1",
+        "Query": "select r1.col from ambiguous_ref_with_source as r1 join ambiguous_ref_with_source",
```

Comment:
Not really sure why the switch from multi-table in `FROM` to a `JOIN`. I think it's fine though?

Reply:
when the planner now is able to see that all the tables used come from the same unsharded keyspace, it shortcuts the planning early in the process and just passes on the query the user wrote.

https://github.com/vitessio/vitess/blob/main/go/vt/vtgate/planbuilder/gen4_planner.go#L187